File: py/latkerlo_jvotci/exceptions.py
"""
Copyright (c) 2023-2024 latkerlo (https://github.com/latkerlo)
Licensed under the MIT License
"""


class DecompositionError(Exception):
    def __init__(self, message=""):
        super().__init__(message)


class InvalidClusterError(Exception):
    def __init__(self, message=""):
        super().__init__(message)


class NoLujvoFoundError(Exception):
    def __init__(self, message=""):
        super().__init__(message)


class NonLojbanCharacterError(Exception):
    def __init__(self, message=""):
        super().__init__(message)


class NotBrivlaError(Exception):
    def __init__(self, message=""):
        super().__init__(message)


class NotZihevlaError(Exception):
    def __init__(self, message=""):
        super().__init__(message)

File: js/docs/exceptions.js
/*
Copyright (c) 2023-2024 latkerlo (https://github.com/latkerlo)
Licensed under the MIT License
*/
class DecompositionError extends Error {
    constructor(message) {
        super(message);
        this.name = "DecopositionError";
    }
}
class InvalidClusterError extends Error {
    constructor(message) {
        super(message);
        this.name = "InvalidClusterError";
    }
}
class NoLujvoFoundError extends Error {
    constructor(message) {
        super(message);
        this.name = "NoLujvoFoundError";
    }
}
class NonLojbanCharacterError extends Error {
    constructor(message) {
        super(message);
        this.name = "NonLojbanCharacterError";
    }
}
class NotBrivlaError extends Error {
    constructor(message) {
        super(message);
        this.name = "NotBrivlaError";
    }
}
class NotZihevlaError extends Error {
    constructor(message) {
        super(message);
        this.name = "NotZihevlaError";
    }
}
